Added physical links to the href attribute for elements in the tree

Ruslan-Aleev opened the pull request:
Added physical links to the href attribute for elements in the tree. Without correct links in the href attribute, the middle and right mouse buttons in Windows / Linux do not work. Read more here - https://github.com/modxcms/revolution/issues/14242

### Related issue(s)/PR(s)
https://github.com/modxcms/revolution/issues/14059

Ruslan-Aleev committed:
Added links to the href attribute for elements

Added physical links to the href attribute for elements in the tree.

Ruslan-Aleev committed:
Fixed a new link to a more correct one

OptimusCrime commented:
This looks good! Any idea what `getHref` actually does?

Ruslan-Aleev commented:
I also did not understand what getHref is used for.
The getHref call is in the files:
/manager/assets/ext3/ext-all-debug.js
/manager/assets/ext3/ext-all.js
/manager/assets/modext/modx.jsgrps-min.js
/manager/assets/modext/util/utilities.js

Jako commented:
It does nothing special, if href is set. Otherwise it returns a hash or an empty space depending on the browser:
```
    getHref : function(href){
        return Ext.isEmpty(href) ? (Ext.isGecko ? '' : '#') : href;
    },
```

Jako commented:
If a.page contains an url everywhere and can't be empty, I am fine with the solution. Otherwise surround that thing with `this.getHref(a.page)`

Ruslan-Aleev commented:
@Jako Empty url will be, for example, for images in files.
And what is the fundamental difference between the empty attribute href="" and the href="#" (for empty url or with getHref check)?

ghost commented:
href="#" will cause a jump to the top of the page. https://stackoverflow.com/questions/5637969/is-an-empty-href-valid

Ruslan-Aleev commented:
@Jako It’s probably better to wrap getHref, I just don’t understand what this check gives :)

Ruslan-Aleev commented:
@sottwell No, the page will not move to the top, now all links in the tree with href="#"

Jako commented:
If I am right, an empty href attribute in an a tag creates cursor issues in some browsers (and maybe does not work properly even with javascript). So it is filled with an # for some browsers (not isGecko - whatever that really means in ExtJS).

Jako commented:
ExtJS 3.4 is quite old and has to handle all those nasty browser incompatibilities to that time. Maybe some checks are not needed anymore.

Ruslan-Aleev committed:
Added links for elements in tree with empty check

Check for empty href for some browsers.
Perhaps this check is relevant only for old ExtJS 3.4.

OptimusCrime commented:
@Jako See this: https://en.wikipedia.org/wiki/Gecko_(software)

Jako commented:
@OptimusCrime Thanks. Gecko should be deprecated now, but maybe isGecko means something different in ExtJS (I don't know the test).

OptimusCrime commented:
It should be deprecated indeed. Calls to `isGecko`, `isGecko2` , and `isGecko3` is found all over the source. I found this:

```
        isWebKit = check(/webkit/),

[...]

        isGecko = !isWebKit && check(/gecko/),
        isGecko2 = isGecko && check(/rv:1\.8/),
        isGecko3 = isGecko && check(/rv:1\.9/),
```

It basically sniffs the user agent to see if the browser is Firefox or some other Gecko based browser.  

This test should be irrelevant in 2019.

ilyautkin commented:
PR was tested. Found bug:

Open resource updating page. Hover any link in the tree. All links in tree gets current page url as href attribute:

![1zpun4](https://user-images.githubusercontent.com/3328625/51231366-22668680-1985-11e9-9244-c40e75248f30.png)

Browser - Crome 71.0.3578.80 (64 bit) for Ubuntu

ilyautkin commented:
PR is tested, it works.
Now middle click is available. Awesome =)
